Tests don't pass for go 1.7

There is a missing assembler function - thanks @glycerine

my pleasure -- terrific to chat yesterday! detail of the missing function:

```
=== RUN   ExampleRun
panic: no code for function: sync.runtime_notifyListCheck
```

--- a/interp/external.go
+++ b/interp/external.go
@@ -15,6 +15,7 @@ import (
 	"runtime"
 	"strings"
 	"sync"
+	"sync/atomic"
 	"syscall"
 	"time"
 
@@ -108,6 +109,7 @@ func init() {
 		"runtime.Gosched":                  ext۰runtime۰Gosched,
 		"runtime.init":                     ext۰runtime۰init,
 		"runtime.NumCPU":                   ext۰runtime۰NumCPU,
+		"runtime.NumGoroutine":             ext۰runtime۰NumGoroutine,
 		"runtime.ReadMemStats":             ext۰runtime۰ReadMemStats,
 		"runtime.SetFinalizer":             ext۰runtime۰SetFinalizer,
 		"(*runtime.Func).Entry":            ext۰runtime۰Func۰Entry,
@@ -120,8 +122,9 @@ func init() {
 		"sync.runtime_Semacquire":          ext۰sync۰runtime_Semacquire,
 		"sync.runtime_Semrelease":          ext۰sync۰runtime_Semrelease,
 		"sync.runtime_Syncsemcheck":        ext۰sync۰runtime_Syncsemcheck,
-		"sync.runtime_canSpin":             sync۰runtime۰canSpin, // Elliott
-		"sync.runtime_doSpin":              sync۰runtime۰doSpin,  // Elliott
+		"sync.runtime_canSpin":             sync۰runtime۰canSpin,         // Elliott
+		"sync.runtime_doSpin":              sync۰runtime۰doSpin,          // Elliott
+		"sync.runtime_notifyListCheck":     sync۰runtime۰notifyListCheck, // Elliott
 		"sync.runtime_registerPoolCleanup": ext۰sync۰runtime_registerPoolCleanup,
 		"sync/atomic.AddInt32":             ext۰atomic۰AddInt32,
 		"sync/atomic.AddUint32":            ext۰atomic۰AddUint32,
@@ -173,6 +176,16 @@ func sync۰runtime۰canSpin(fr *frame, args []Ivalue) Ivalue { //i int) bool { /
 }
 func sync۰runtime۰doSpin(fr *frame, args []Ivalue) Ivalue { //) { // Elliott
 	runtime.Gosched()
+	return nil
+}
+
+// NumGoroutine returns the number of goroutines that currently exist.
+func ext۰runtime۰NumGoroutine(fr *frame, args []Ivalue) Ivalue { // Elliott
+	return int(atomic.LoadInt32(&fr.i.goroutines))
+}
+
+// Ensure that sync and runtime agree on size of notifyList.
+func sync۰runtime۰notifyListCheck(fr *frame, args []Ivalue) Ivalue { // Elliott
 	return nil
 }
 

--- a/interp/interp.go
+++ b/interp/interp.go
@@ -54,6 +54,7 @@ import (
 	"reflect"
 	"runtime"
 	"sync"
+	"sync/atomic"
 
 	"go/types"
 	"golang.org/x/tools/go/ssa"
@@ -88,6 +89,7 @@ type interpreter struct {
 	rtypeMethods       methodSet             // the method set of rtype, which implements the reflect.Type interface.
 	runtimeErrorString types.Type            // the runtime.errorString type
 	sizes              types.Sizes           // the effective type-sizing function
+	goroutines         int32                 // atomically updated
 
 	externals *Externals // per-interpreter externals (immutable)
 	/*
@@ -318,11 +320,13 @@ func visitInstr(fr *frame, instr ssa.Instruction) continuation {
 
 	case *ssa.Go:
 		fn, args := prepareCall(fr, &instr.Call)
+		atomic.AddInt32(&fr.i.goroutines, 1)
 		go func() {
 			defer func() {
 				fr.i.userStackIfPanic()
 			}()
 			call(fr.i, nil, instr.Pos(), fn, args)
+			atomic.AddInt32(&fr.i.goroutines, -1)
 		}()
 
 	case *ssa.MakeChan:
@@ -746,10 +750,11 @@ func Interpret(mainpkg *ssa.Package, mode Mode, sizes types.Sizes, filename stri
 		ext = NewExternals()
 	}
 	i := &interpreter{
-		prog:    mainpkg.Prog,
-		globals: make(map[ssa.Value]*Ivalue),
-		mode:    mode,
-		sizes:   sizes,
+		prog:       mainpkg.Prog,
+		globals:    make(map[ssa.Value]*Ivalue),
+		mode:       mode,
+		sizes:      sizes,
+		goroutines: 1,
 
 		//constants: make(map[*ssa.Const]Ivalue),
 		externals:   ext,

--- a/ssainterp.go
+++ b/ssainterp.go
@@ -7,10 +7,9 @@ package ssainterp
 
 import (
 	"fmt"
+	"github.com/go-interpreter/ssainterp/interp"
 	"go/build"
 	"io"
-
-	"github.com/go-interpreter/ssainterp/interp"
 	//"./interp"
 
 	"go/types"

